### ISSUE TITLE
Mapping of start and end dates to FHIR  

### ISSUE THREAD

**Issue by ArdonToonstra** [Nictiz/Nictiz-R4-zib2020#52]
**How to map zib concepts StartDate and EndDates to FHIR**

A number of zibs have a concept for a StartDate and EndDate (described in various different conceptnames, e.g. ProcedureStartDate and ProcedureEndDate) both based on a Timestamp datatype. The definition text of those concepts may have additional requirements on how to interpret them. Zibs don&#39;t have a &#39;period&#39; datatype but do have a part zib named TimeInterval. This is however not applied to all zibs (mostly only within medication-related zibs).

FHIR does have Period datatype, consisting of a `.start` and `.end` DateTime element. A Period contains additional meaning, for example a missing .end element implies an ongoing period. Which in turn affects the functioning of searches.

So far it seems we should look at this per zib to see what additional requirements the zib makes and if FHIR can natively support a Period and a DateTime. This issue indicates the possible solutions and tries to keep track of the solutions chosen per zib.

Potential solutions

1. Place the mapping on the value[x] element that contains both Period and DateTime option. Describe when to use Period and Datetime.
2. Place the StartDate on value[x] and the EndDate more explicitly on `valuePeriod.end`. Describe the usage of StartDate. Makes the profile likely a lot less readable.
3. Map both concepts to the Period datatype (StartDate to Period.start and EndDate to Period.end). To exchange an instantaneous moment, use a `Period.start` and `Period.end` with a difference of one second.
4. Other?

**Per zib overview
|     Zib elements                                                               |     Describes usage as a period and datetime?    |     Maps to a value[x] with both Period and DateTime                                     |     Potential solution?                    |
|--------------------------------------------------------------------------------|--------------------------------------------------|------------------------------------------------------------------------------------------|--------------------------------------------|
|     Procedure (ProcedureStartDate and ProcedureEndDate)                        |     Yes                                          |     Yes    (Procedure.perfomed[x] and    ServiceRequest. occurrence[x])                  |     Option 1                               |
|     Encounter (StartDateTime and EndDateTime)                                  |     Yes, but describes it slightly indirect      |     No    (Encounter.period)                                                             |     Option 3     ([Related Zulip chat](https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Encounter.2Eperiod))    |
|     Alert (StartDateTime and EndDateTime)                                      |     No                                           |     No (Flag.period)                                                                     |     Option 3                               |
|     EpisodeOfCare (StartDate and EndDate)                                      |     No                                           |     No (EpisodeOfCare.period)                                                            |     Option 3                               |
|     TobaccoUse (StartDate and StopDate)                                        |     No                                           |     Yes (Observation.effective[x])                                                       |     Option 1                               |
|     AlcoholUse (StartDate and StopDate)                                        |     No                                           |     Yes (Observation.effective[x])                                                       |     Option 1                               |
|     DrugUse (StartDate and StopDate)                                           |     No                                           |     Yes (Observation.effective[x])                                                       |     Option 1                               |
|     CareAgreement (StartDate and EndDate)                                      |     No                                           |     Mogelijk (CarePlan… scheduled[x])                                                    |     Option 1                               |
|     NursingIntervention( ProcedureStartDateTime   and ProcedureEndDateTime)    |     Yes, but vaguely                             |     Yes    (Procedure.perfomed[x] and    ServiceRequest. occurrence[x])                  |     Option 1                               |
|     FreedomrestrictingIntervention   (Start and End)                           |     Yes                                          |     Yes (Procedure.perfomed[x])                                                          |     Option 1                               |
|     LegalSituation (DateStart and DateStop)                                    |     No                                           |     No, has two different elements (Condition.onset[x]   and Condition.abatement[x])     |     n/a                                    |
|     Problem (ProblemStartDate and ProblemEndDate)                              |     No, two different concepts                   |     No, has two different elements (Condition.onset[x]   and Condition. abatement[x])    |     n/a                                    |

**Comment by pieter-edelman-nictiz** [comment]
The zibs that are marked "No" in the second column don't need to be mapped to DateTime, or am I missing something? Eg. in TobaccoUse/AlcoholUse/DrugUse StartDate and StopDate seem to have the same meaning as a proper Period.

**Comment by ArdonToonstra** [comment]
@pieter-edelman-nictiz , fair point. The TobaccoUse/AlcohlUse/DrugUse do seem to have the same meaning as a Period. Although it is not explicitly stated by the zib: you don't know if there is a difference between an unknown StopDate or an ongoing period for sure. 
But I agree that these may distract from the real cases, so I will break up the table in two.

**Comment by ArdonToonstra** [comment]
In zib-procedure and zib-FreedomRestrictingIntervention a 4th option has been proposed: make another type slice for a DateTime and map the StartDate to it. This seems a more explicit/specific solution.

**Comment by ArdonToonstra** [comment]
@pieter-edelman-nictiz , ik heb een poging gewaagd dit te documenteren in de profiling guidelines: https://informatiestandaarden.nictiz.nl/wiki/FHIR:Vissue-github-52_FHIR_Profiling_Guidelines_R4#Mapping_start_and_end_dates_to_profiles 

Daarnaast de profielen nagelopen. Het staat nu goed bij Procedure-event, Procedure-request, FreedomRestrictingIntervention en Encounter. Vraag me af of jij vind of er ook iets moet staan bij `Careplan.period`, als bij `Encounter.period` in NursingIntervention profiel.

**Comment by pieter-edelman-nictiz** [comment]
@ArdonToonstra ik heb een beetje aan de tekst geschaafd; kijk maar of je het ermee eens bent. Zullen we ook een standaardtekstje opnemen voor Period-als-moment?

Je triggert me even op NursingIntervention. Volgens [ZIB-1502](https://bits.nictiz.nl/browse/ZIB-1502) kan de zib zowel een gebeurtenis als een gepland iets als een voornemen vertegenwoordigen. Dat zou betekenen dat we er maar liefst _drie_ profielen voor moeten maken.

**Comment by ArdonToonstra** [comment]
@pieter-edelman-nictiz , eens met de aangepaste tekst. Dank.
Een standaardtekstje voor Period-als-moment lijkt me ook goed. Momenteel staat het volgende in de comment op `Encounter.period` (voor de zib Encounter)
>If only one DateTime is known the `period.start` and `period.end` hold equal values to avoid expressing an ongoing period.

Dat is misschien iets te weinig?
> When only one start date or end date is known which is not part of expressing a period, the `.period.start` and `.period.end` hold equal values to avoid expressing an ongoing period.

Mm, ZIB-1502 is inderdaad vervelend! Niet direct een antwoord op.
